searx/templates/__common__/opensearch.xml

```diff
@@ -6,9 +6,9 @@
   <Image>{{ urljoin(host, url_for('static', filename='img/favicon.png')) }}</Image>
   <LongName>searx metasearch</LongName>
   {% if opensearch_method == 'get' %}
-    <Url rel="results" type="text/html" method="get" template="{{ url_for('search', _external=True) }}?q={searchTerms}"/>
+    <Url rel="results" type="text/html" method="get" template="{{ host }}search?q={searchTerms}"/>
```

Comment:
Is this URL going to be correct always? Is it required for host to include `/`? I haven't seen a check for this in the code.

Reply:
Good question I had also .. I had this trailing slash issue in my engine, have a look at the existing *autocompleter* implementation:

https://github.com/searx/searx/blob/11070e57449caf86fd6087116e24bed8bea54e63/searx/templates/__common__/opensearch.xml#L16

So I added a trailing slash to the base_url like it was given in the comment `https://your.custom.host/location/`

https://github.com/searx/searx/blob/11070e57449caf86fd6087116e24bed8bea54e63/searx/settings.yml#L26

What do you think, does it make sense to extend ..

https://github.com/searx/searx/blob/11070e57449caf86fd6087116e24bed8bea54e63/searx/webapp.py#L272-L280

and check if there is a trailing slash?

Reply:
> Is this URL going to be correct always? Is it required for host to include /? I haven't seen a check for this in the code.

@kvch I normalized the opensearch.xml template to make use of the `url_for(..., _external=True)` function from Flask (aka werkzeug). This implementations handles a leading `/` well, it doesn't matter if three is a leading `/` or not, both configurations of `base_url` are now valid..